identity/handler.go

```diff
@@ -524,3 +561,84 @@ func (h *Handler) delete(w http.ResponseWriter, r *http.Request, ps httprouter.P
 
 	w.WriteHeader(http.StatusNoContent)
 }
+
+// swagger:route PATCH /admin/identities/{id} v0alpha2 adminPatchIdentity
+//
+// Partially updates an Identity's field
+//
+// This endpoint updates an identity. Only the fields that are supplied in the payload are updated.
+//
+// Note: If the `state` of the identity is updated the `stated_changed_at` field is set to the current timestamp.
+//
+// Learn how identities work in [Ory Kratos' User And Identity Model Documentation](https://www.ory.sh/docs/next/kratos/concepts/identity-user-model).
+//
+//     Consumes:
+//     - application/json
+//
+//     Produces:
+//     - application/json
+//
+//     Schemes: http, https
+//
+//     Security:
+//       oryAccessToken:
+//
+//     Responses:
+//       200: identity
+//       400: jsonError
+//       404: jsonError
+//       409: jsonError
+//       500: jsonError
+func (h *Handler) patch(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	var ur AdminPatchIdentityBody
+	if err := h.dx.Decode(r, &ur,
+		decoderx.HTTPJSONDecoder()); err != nil {
+		h.r.Writer().WriteError(w, r, err)
+		return
+	}
+
+	id := x.ParseUUID(ps.ByName("id"))
+	identity, err := h.r.PrivilegedIdentityPool().GetIdentityConfidential(r.Context(), id)
+	if err != nil {
+		h.r.Writer().WriteError(w, r, err)
+		return
+	}
+
+	if ur.SchemaID != "" {
+		identity.SchemaID = ur.SchemaID
+	}
+
+	if ur.State != "" && identity.State != ur.State {
+		if err := ur.State.IsValid(); err != nil {
+			h.r.Writer().WriteError(w, r, errors.WithStack(herodot.ErrBadRequest.WithReasonf("%s", err).WithWrap(err)))
+			return
+		}
+
+		stateChangedAt := sqlxx.NullTime(time.Now())
+
+		identity.State = ur.State
+		identity.StateChangedAt = &stateChangedAt
+	}
+
+	if len(ur.Traits) > 0 {
+		identity.Traits = []byte(ur.Traits)
+	}
```

Comment:
The nested traits are not patched but replaced here. We should mention that in the description.

Reply:
True, I added a comment to the documentation comment.

It might make sense to support patching here, though I think that may cause a lot of confusion. Maybe something for later on...

Reply:
Hm but that will be a potentially breaking change. I'd rather decide now to either support it or not here. We can also add a separate endpoint for that at a later point.

Reply:
I see, if I understand that part correctly, the `traits` field could be any arbitrary data type, e.g. a string? That means someone could put an XML string in there, which would break the patching ability here. So in order to completely support patching on that field, there would need to be a lot of logic involved. Also, I am not sure if there is support for deep merging of JSON fields out of the box here. Which means, that needs to be implemented.

I think an extra endpoint for updating traits could be a good solution later on.